partiql-planner/src/testFixtures/resources/inputs/basics/case.sql

```diff
@@ -292,6 +292,56 @@ CASE t_item.t_string
     ELSE t_item.t_any
 END;
 
+--#[case-when-35]
+-- type: decimal(10,5)|null
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_decimal_10_5
+    ELSE null
+END;
+
+--#[case-when-36]
+-- type: decimal(10,5)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_decimal_10_5
+    ELSE t_item.t_decimal_5_3
+END;
+
+--#[case-when-37]
+-- type: varchar(10)|null
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_varchar_10
+    ELSE null
+END;
+
+--#[case-when-38]
+-- type: varchar(10)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_varchar_10
+    ELSE t_item.t_varchar_5
+END;
+
+--#[case-when-39]
+-- type: char(10)|null
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_char_10
+    ELSE null
+END;
+
+--#[case-when-40]
+-- type: char(10)
+CASE t_item.t_string
+    WHEN 'a' THEN t_item.t_char_10
+    ELSE t_item.t_char_5
+END;
+
+--#[case-when-41]
+-- type: varchar(10)
```

Comment:
I'm curious what would happen if you had a `CHAR(x)` and `VARCHAR(y)` where `x` > `y`?

Reply:
VARCHAR(x). 

It seems to be common practice and sane approach to envision varchar having higher type precedence than char. SQL Server [example](https://learn.microsoft.com/en-us/sql/t-sql/data-types/data-type-precedence-transact-sql?view=sql-server-ver16). 

It also makes sense to preserve the length constraint to be able to hold the largest possible data.

Reply:
Ok cool. That lines up w/ my expectation for the behavior.